htdocs/compta/bank/various_payment/card.php

```diff
@@ -390,20 +432,28 @@
 	}
 
 	// Type payment
-	print '<tr><td>';
-	print $form->editfieldkey('PaymentMode', 'selectpaymenttype', '', $object, 0, 'string', '', 1).'</td><td>';
-	$form->select_types_paiements($paymenttype, "paymenttype");
-	print '</td></tr>';
+	print '<tr><td><span class="fieldrequired">'.$langs->trans('PaymentMode').'</span></td><td>';
+	$form->select_types_paiements($paymenttype, 'paymenttype', '', 2);
```

Comment:
You changed the value returned by the select by returning code instead of id but i did not found modification of code that used the value posted. 
And code to get the value is 
GETPOST('paymenttype', 'int')
So this seems to introduce a bug. Id must be prefered to code when we have an id for key.